### PR TITLE
Allow to use the same template on Search Module to results and no res…

### LINKED ISSUE #1319: Allow `results` and `no_results` on search to use the same template
**Is your feature request related to a problem? Please describe.**
The need to have a template for search with and without results makes us to code more than necessary.

**Describe the solution you'd like**
Make `exp:search_results` to accept `no_results` conditional.

**Teachability, Documentation, Adoption, Migration Strategy**

It can be done at [`mod.search.php`](https://github.com/ExpressionEngine/ExpressionEngine/blob/7c8e99473ea9a03595cd7eb6ffe7f07d363f73ad/system/ee/ExpressionEngine/Addons/search/mod.search.php#L1071):

```
if ($query->num_rows() == 0 or $query->row('total_results') == 0) {
    return ee()->TMPL->no_results();
```

- [x] I am capable and would like to work on implementation of this feature if it is considered.

### PR DESCRIPTION
**Pull request by robsonsobral**
## Overview

Allow to use the same template on Search Module to results and no results

Resolves [#1319](https://github.com/ExpressionEngine/ExpressionEngine/issues/1319).

## Nature of This Change

- [ ] 🐛 Fixes a bug
- [x] 🚀 Implements a new feature
- [ ] 🛁 Refactors existing code
- [ ] 💅 Fixes coding style
- [ ] ✅ Adds tests
- [ ] 👽 Adds new dependency
- [ ] 🔥 Removes unused files / code
- [ ] 🔒 Improves security 

## Is this backwards compatible?

- [x] Yes
- [ ] No

## Documentation

User Guide Pull Request: https://github.com/ExpressionEngine/ExpressionEngine-User-Guide/pull/345

### CODE REVIEW

File: system/ee/ExpressionEngine/Addons/search/mod.search.php [hunk 1]
```diff
@@ -1068,12 +1072,21 @@ public function search_results()
...
-        if ($query->num_rows() == 0 or $query->row('total_results') == 0) {
-            // This should be impossible as we already know there are results
+        if ($query->num_rows() == 0) {
             return ee()->output->show_user_error(
                 'general',
                 array(lang('invalid_action'))
             );
+        } elseif ($query->row('total_results') == 0) {
+            // this works if we use the same template for results and no results
+            if ($query->row('result_page') === $query->row('no_result_page')) {
```
Comment (intoeetive) [inline system/ee/ExpressionEngine/Addons/search/mod.search.php:1117]: The `no_result_page` column does not exist, so can't be used here. Did you mean `$this->_meta['no_result_page']`?

File: system/ee/ExpressionEngine/Addons/search/upd.search.php [hunk 2]
```diff
@@ -149,6 +149,10 @@ public function update($current = '')
                 ->update('actions', array('csrf_exempt' => 1));
         }
 
+        if (version_compare($current, '2.3', '<')) {
+            ee()->smartforge->add_key('search', 'no_result_page');
```
Comment (intoeetive) [inline system/ee/ExpressionEngine/Addons/search/upd.search.php]: This column does not exist, so key can't be added

**Review by intoeetive — CHANGES_REQUESTED** [review]
@robsonsobral This looks like nice addition, but some changes are needed - please see the comments.

Also, can you please merge the latest `6.dev` - that will fix the tests run

File: system/ee/ExpressionEngine/Addons/search/upd.search.php (hunk 2 above)
Comment (robsonsobral) [reply system/ee/ExpressionEngine/Addons/search/upd.search.php]: @intoeetive , how could I add a new column to the database?

Comment (intoeetive) [reply system/ee/ExpressionEngine/Addons/search/upd.search.php]: You would need to use `ee()->smartforge->add_column`
But is there a need for this column at all? Could it not be always fetched from tag parameter instead?

Comment (robsonsobral) [reply system/ee/ExpressionEngine/Addons/search/upd.search.php]: Thank you, @intoeetive . Yeah, it's need. The values of tag parameters are present only on the form tag, not on the results one.

**Review by intoeetive — DISMISSED** [review]
@robsonsobral I've added some automated tests around this addition and it seems to work well.

Thank you!

**Review by TomJaeger — DISMISSED** [review]

**Review by TomJaeger — DISMISSED** [review]

**Review by intoeetive — APPROVED** [review]

**Review by matthewjohns0n — APPROVED** [review]

**Pull request merged**